src/transaction/log_recovery_redo.hpp

```diff
@@ -736,6 +714,32 @@ void log_rv_redo_record_sync (THREAD_ENTRY *thread_p, log_rv_redo_context &redo_
       pgbuf_set_lsa (thread_p, rcv.pgptr, &record_info.m_start_lsa);
       // rcv pgptr will be automatically unfixed at the end of the parent scope
     }
+
+}
+
+template <typename T>
+void log_rv_redo_record_sync (THREAD_ENTRY *thread_p, log_rv_redo_context &redo_context,
+			      const log_rv_redo_rec_info<T> &record_info, const VPID &rcv_vpid)
+{
+#if !defined(NDEBUG)
+  if (log_Gl.rcv_phase != LOG_RESTARTED)
+    {
+      // bit of debug code to ensure that, should this code be executed asynchronously, within the same page,
+      // the lsa is ever-increasing, thus, not altering the order in which it has been added to the log in the first place
+      log_Gl_recovery_redo_consistency_check.check (rcv_vpid, record_info.m_start_lsa);
+    }
+#endif
+
+  LOG_RCV rcv;
```

Comment:
after declaring `rcv` and before calling the function, still use the `scope_exit` with the functor, but replace the handwritten unfix with `pgbuf_unfix_and_init` macro